### PR TITLE
Kth maximum element in array in C++

### LINKED ISSUE #162: Kth minimum/maximum element in an array
Hi,
I am a GSSOC'21 participant.
I want to add the various solutions of finding the Kth smallest/largest element in an array. A simple and straightforward that everyone probably knows is simple sorting that takes O(N log N) time.  I will discuss the various implementations in linear time complexity.
* My code will be in C++
* I will add proper test cases explaining every approach to the problem

C++
- [ ] Minimum
- [ ] Maximum

Java
- [ ] Minimum
- [ ] Maximum

Python
- [ ] Minimum
- [ ] Maximum

Please assign me this issue @nakul-19 @plazzy99 @rudrakshi99

### PR DESCRIPTION
**Pull request by vedant-jain03**
fixes #162 
-> Added C++ Code
-> Updated Readme file
@tarun26091999 @plazzy99 @rudrakshi99 @raksha009

### CODE REVIEW

File: Code/C++/Kth_maximum_in_array.cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    int arr[n];
+    for(int i=0;i<n;i++)
+    {
+        cin>>arr[i]; // Input the elements
+    }
+    int k;
+    cin>>k; // Input K
+    cout<<Kth_max_element(arr,n,k)<<"\n";
+}
+
```
Comment (krishna-NIT) [inline Code/C++/Kth_maximum_in_array.cpp:28]: Include sort Function which is mentioned above

**Review by krishna-NIT — CHANGES_REQUESTED** [review]

**Review by vedant-jain03 — COMMENTED** [review]
Updated with Sort function

File: Array/readme.md [hunk 2]
```diff
@@ -27,5 +27,4 @@ To create an array, you need to known the length (or size) of the array in advan
  * Insertion_at_start ----> [C++](/Code/C++/insertion_at_start.cpp) 
  * Kadane's Algorithm ----> [C++](/Code/C++/kadane_algo.cpp) 
  * Repeating and Missing Number ----> [C++](/Code/C++/repeating_and_missing_number.cpp) 
- * Peak element in array using linear search ----> [C++](/Code/C++/peak_value_linear_search.cpp)
-
```
Comment (tarun26091999) [inline Array/readme.md:31]: you cannot remove these lines.

**Review by tarun26091999 — CHANGES_REQUESTED** [review]
this is not the most optimized approach. You took this long to raise the PR and still the solution nowhere near to the required quality. Also there are conflicts in code base.

**Comment by vedant-jain03** [comment]
> this is not the most optimized approach. You took this long to raise the PR and still the solution is nowhere near to the required quality. Also there are conflicts in code base.

Should I use heap to implement this?

**Comment by rudrakshi99** [comment]
> > this is not the most optimized approach. You took this long to raise the PR and still the solution is nowhere near to the required quality. Also there are conflicts in code base.
> 
> Should I use heap to implement this?

Yes

**Pull request closed without merging**